File: fron-end/src/app/cotizacion/cotizacion.service.ts
import { Injectable } from '@angular/core';
import {Cotizacion} from './cotizacion';
import { Observable } from 'rxjs';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class CotizacionService {

    private httpOptions =   {
      headers: new HttpHeaders ({"Content-Type":"application/json"})
    }

    private url = environment.API_URL + "/cotizacion";
 
    constructor(
     private http: HttpClient 
    ){ }


  //Create
  public save(cotizacion: Cotizacion): Observable<Cotizacion>
  {
    return this.http.post<Cotizacion>
    (this.url+"/save",cotizacion, this.httpOptions);
  }


  //Read
  public findById(id: number): Observable<Cotizacion>
  {
    return this.http.get<Cotizacion>(this.url+"/id", this.httpOptions );
  }

  //Update
  public update(cotizacion: Cotizacion ): Observable<Cotizacion>
  {
    return this.http.put<Cotizacion>(this.url+"/update", cotizacion, this.httpOptions);
    
  }

  //Delete
  public deleteById(id:number): Observable<Cotizacion>
  {
    return this.http.delete<Cotizacion>(this.url+"/deleteById/"+id, this.httpOptions);
  }

  //FinalAll
    public findAll(): Observable<Cotizacion[]>
    {
        return this.http.get<Cotizacion[]>(this.url+"/findAll" , this.httpOptions)
    }

}

File: back-end/src/main/java/CRM4RTONOCTURNOA/CRM/service/CotizacionService.java
package CRM4RTONOCTURNOA.CRM.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import CRM4RTONOCTURNOA.CRM.entity.Cotizacion;
import CRM4RTONOCTURNOA.CRM.repository.CotizacionRepository;

@Service
public class CotizacionService {

    @Autowired
    CotizacionRepository cotizacionRepository;

    //Create y Update
    public Cotizacion save(Cotizacion cotizacion)
    {
        return cotizacionRepository.save(cotizacion);
    }

    //Read
    public Cotizacion findById(Long id)
    {
        return cotizacionRepository.findById(id).get();
    }

    //Delete
    public void deleteById(Long id)
    {
        cotizacionRepository.deleteById(id);
    }

    public List<Cotizacion> findAll()
    {
        return cotizacionRepository.findAll();
    }


}
